src/components/Avatar.js

```diff
@@ -40,14 +46,50 @@ const defaultProps = {
     fallbackIcon: Expensicons.FallbackAvatar,
 };
 
-class Avatar extends PureComponent {
+class Avatar extends Component {
     constructor(props) {
         super(props);
         this.state = {
             imageError: false,
         };
     }
 
+    shouldComponentUpdate(nextProps, nextState) {
+        if (
+            lodashGet(this.props.network, 'isOffline')
+                  && !lodashGet(nextProps.network, 'isOffline')
+        ) {
+            return true;
+        }
+        if (this.props.source !== nextProps.source) {
+            return true;
+        }
+        if (this.state.imageError !== nextState.imageError) {
+            return true;
+        }
+
+        if (this.props.fill !== nextProps.fill) {
+            return true;
+        }
+        if (this.props.size !== nextProps.size) {
+            return true;
+        }
+
+        return false;
+    }
```

Comment:
Why do we need this?

Reply:
@parasharrajat I use `withNetwork()` to check the current internet status, but I also don't want the Avatar to update when the internet goes from online to offline, because it will trigger `onError` and make the Avatar render the placeholder. 
```javascript
      // Only update the component if the internet goes from offline to online
        if (
            lodashGet(this.props.network, 'isOffline')
                  && !lodashGet(nextProps.network, 'isOffline')
        ) {
            return true;
        }
```
Since we need to handle the `shouldComponentUpdate` ourself, the last 4 conditions to prevent redundant render.
```javascript
        if (this.props.source !== nextProps.source) {
            return true;
        }
        if (this.state.imageError !== nextState.imageError) {
            return true;
        }

        if (this.props.fill !== nextProps.fill) {
            return true;
        }
        if (this.props.size !== nextProps.size) {
            return true;
        }

        return false;
```

Reply:
@hungvu193 this will also prevent rerenders if `imageStyles` and `containerStyles` changes which is I think important if we need this component to be reusable and flexible

Reply:
For now I think that's the best to handle it.

Reply:
> but I also don't want the Avatar to update when the internet goes from online to offline, 
 
What is the issue with that? I don't see any issue.

Reply:
https://user-images.githubusercontent.com/16502320/203202782-e047903a-04c5-4935-8163-4d2a695e84ae.mov

Here's the example, without using the `componentDidupdate`.
I think I can remove `componentDidupdate` in `Avatar.js` if we can pass a prop `isOffline` into it, by doing this way we can avoid using withNetwork and we don't need to use `componentDidupdate` anymore.

Reply:
I tested it on chrome and I didn't see this issue. Which browser are you using?

Reply:
Using shouldComponentUpdate and componentDidUpdate also feels wrong to me. But I'd need to test things myself in order to provide an alternative

Reply:
I do have an alternative. That might fit well. Let me test it first.

Reply:
An alternative might be to use use a transformValue function on `withNetwork` and get only isOffline value as a prop. You can use pass `propName` to change the prop from `network` to isOffline`. 


But I still can't see the issue. I don't know how you are testing but I see no issue. For me, the image never switches when I go offline.  When an image is loaded and even the prop changes, the image should remain loaded because the source value never changed. Prop changes will re-render the component but it does not mean the image will reset. Only requests that are in flight will fail which is expected. 


@hungvu193 Can you share the logical reasoning on why you think it fails? Do you mind sharing the exact steps?

Reply:
@parasharrajat step to reproduce:
1. check out main branch, pull latest code.
2. Disconnect the internet.
3. Choose another report the image didn't load.
4. Connect the internet again.
5. Notice that the avatar isn't loaded.
Video:

https://user-images.githubusercontent.com/16502320/204071762-c1502eeb-3dd3-415c-8136-2745bc37d5a0.mov

Reply:
Are these steps for the main issue or the issue you mentioned in https://github.com/Expensify/App/pull/12808#discussion_r1025943199? 

I am asking for the details on https://github.com/Expensify/App/pull/12808#discussion_r1025943199

> I use withNetwork() to check the current internet status, but I also don't want the Avatar to update when the internet goes from online to offline, because it will trigger onError and make the Avatar render the placeholder.

Reply:
These steps for the main issue.
In main branch when you disconnect the internet the avatar turn to grey but when you connect the internet again it doesn't reload the internet, so in my PR I prevent that thing.
So my idea is only update the component if it's error and the internet connection goes from offline to online.
we can also add the check with imageStyles if needed.

Reply:
I think you misunderstood my question. I agree with what you said above but I am asking about the purpose of `shouldComponentUpdate`. 

If I remove `shouldComponentUpdate`,https://github.com/Expensify/App/pull/12808#discussion_r1033074788 will still work. 


Now you mentioned that we need `shouldComponentUpdate` for an issue https://github.com/Expensify/App/pull/12808#discussion_r1025943199
> I use withNetwork() to check the current internet status, but I also don't want the Avatar to update when the internet goes from online to offline, because it will trigger onError and make the Avatar render the placeholder.


I want to know the reproductions steps for this issue. I can not reproduce this. My logical reasoning is https://github.com/Expensify/App/pull/12808#discussion_r1032584031

Reply:
I tried testing the same and it was not the case for me. Let me test it again.

Reply:
> @hungvu193 Please merge the main. There are some important changes.

@parasharrajat Updated :D

Reply:
@yuwenmemon Can you please test this and take us out of this dilemma?

Reply:
Suuuure will take a look!

Reply:
I'm seeing the same as @parasharrajat.

I would tend to think that `componentDidUpdate` by itself _should_ be fine too, based on other examples I've seen in code by Expensify internal engineers: https://github.com/Expensify/App/blob/d8eb4c6ee32b2bbab93e7d383edec8a165986388/src/pages/workspace/WorkspaceInvitePage.js#L96-L100

Reply:
Disable caching is checked, please uncheck and try @hungvu193

Reply:
Yeah, so the difference between behavior is happening because of disabling the cache. I will run some tests and the next thing to decide is whether it is necessary to handle this case.

Reply:
> Disable caching is checked, please uncheck and try @hungvu193
Oh god. Thank you, I totally forgot about that. Since I worked in many project I always disable the cache.
@parasharrajat let me know if we need to handle this case, otherwise `componentDidUpdate` is ok.

Reply:
@yuwenmemon Should we be worried about that cache issue in this PR? I think there was an issue to improve image caching. May be https://github.com/Expensify/App/issues/10792

Reply:
I don't think I'm worried about it. It seems like a pretty edge-case thing to have the setting disabled, right? I don't think we need to optimize for that in the code.

Reply:
@hungvu193 let's remove shouldComponentUpdate and back to PureComponent

Reply:
Cool, I updated it.